### PR TITLE
Reg 67/clean up reducers

### PR DESCRIPTION
**Pull request by arianaolson419**
https://olinjs.atlassian.net/browse/REG-67

Waiting on the Course Planner page to finish restructuring its state, but everything else is nicer!
image

### CODE REVIEW

**Comment by yunhsincynthiachen** [comment]
Checking it out

File: src/reducers/Progress.jsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
+import { combineReducers } from 'redux';
+
+// Array reducing funcions
```
Comment (logandavis) [inline src/reducers/Progress.jsx:3]: nit: typo (funcions -> functions)

File: src/reducers/Progress.jsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+const progressInitialState = {
+  courses: [],
+  creditTotal: 0,
+};
+
+const mathCourses = (state = progressInitialState, action) => {
+  switch (action.type) {
+    case 'GET_COMPLETED_COURSES': {
+      const courses = action.completedCourses
+        .filter(course => course.registrarId.substring(0, 3) === 'MTH');
```
Comment (yunhsincynthiachen) [inline src/reducers/Progress.jsx:43]: Awesome way of doing it!

File: src/reducers/Student.jsx [hunk 3]
```diff
@@ -0,0 +1,163 @@
+import { combineReducers } from 'redux';
```
Comment (logandavis) [inline src/reducers/Student.jsx:1]: Overall I think that the new state shape is much nicer. If the stuff that you've tried appears to work, I say we ship this. My one significant thought is that it seems like the reducers which are dependent on database calls should probably have cases to handle when the database request fails or times out. But the "nothing" behavior is a sane default, so we're probably good.

**Review by logandavis — APPROVED** [review]

**Review by yunhsincynthiachen — APPROVED** [review]
Everything looks awesome! I noticed we're missing logging out functionality. Maybe make sure to keep that in mind with the reducers.

**Pull request merged**